src/linker/Linker/Driver.cs

```diff
@@ -675,19 +675,23 @@ public bool Run (ILogger customLogger = null)
 			try {
 				p.Process (context);
 			} catch (Exception ex) {
-				Console.Error.WriteLine ("IL Linker has encountered an unexpected error. Please report the issue at https://github.com/mono/linker/issues");
-				if (ex is LinkerFatalErrorException le) {
-					context.LogMessage (le.MessageContainer);
-					if (le.InnerException != null)
-						Console.Error.WriteLine (le.InnerException);
+				if (ex is LinkerFatalErrorException lex) {
+					context.LogMessage (lex.MessageContainer);
 				} else {
-					Console.Error.WriteLine (ex);
+					string exceptionMessage = ex.Message ?? string.Empty;
+					context.LogMessage (MessageContainer.CreateErrorMessage ($"IL Linker has encountered an unexpected error. Please report the issue at https://github.com/mono/linker/issues \n{exceptionMessage}", 1012));
 				}
+
+				string innerMessage = ex.InnerException?.Message ?? null;
```

Comment:
I think we need to print whole .ToString () in this case somewhere instead. Showing one level of inner-exception does not really help